### PR TITLE
apply-filter functionality

### PR DESCRIPTION
**Pull request by Alexandru-Bulai**
# PR 06-02-24
## Overview
In this PR we add the main functionality to the apply button (changes still needed)
## Key Changes :
 1) gallery.js :
    - We toogle an active class on each singleAnimalFilter div 
    - When we click the apply button we prevent the browser to interfire by refreshing the page or somenthing else
    - When we click the button we retreive the data that the user selected and display it when we click apply
2) gallery.html :
    - Updated the rating system from name to have data-rating
3) gallery-dom.js :
    - Added gallery-dom.js file were we are going to write all the DOM related code to avoid conflic with jest
    - Can be tested separately with e2e playwright later on 
# Additional notes
The apply functionality is far from being ok but this is only the first step, after that improvements will come into place

### CODE REVIEW

File: public/JavaScript/gallery.js [hunk 1]
```diff
@@ -1,48 +1,34 @@
...
-//     applyFilters.addEventListener('click', showSelectedAnimals)
-//   } else {
-//     console.error('Apply filters button not found')
-//   }
-//   setSelectedAnimalType()
-// })
+document.addEventListener('DOMContentLoaded', function () {
+  // Event listeners for animal filter buttons
+  document.querySelectorAll('#singleAnimalFilter div[data-type]').forEach(filterDiv => {
+    filterDiv.addEventListener('click', function () {
```
Comment (vladiiancu) [inline public/JavaScript/gallery.js]: The name `filterDiv` is tied too much to implementation.

File: public/JavaScript/gallery.js [hunk 2]
```diff
@@ -1,48 +1,34 @@
...
+  applyFiltersButton.addEventListener('click', function (event) {
+    event.preventDefault()
+
+    // Retrieve selected filter values
+    const genderFilter = document.querySelector('input[name="gender"]:checked')?.value
+    const typeFilters = document.querySelectorAll('#singleAnimalFilter div[data-type].active')
+    const ratingFilter = document.querySelector('.star-rating input[type="radio"]:checked')?.getAttribute('data-rating')
+
+    // Prepare the selected filters object
+    const userSelectedFilters = {
```
Comment (vladiiancu) [inline public/JavaScript/gallery.js]: Nice

**Review by vladiiancu — DISMISSED** [review]

File: public/JavaScript/gallery.js [hunk 3]
```diff
@@ -1,48 +1,34 @@
...
-//     applyFilters.addEventListener('click', showSelectedAnimals)
-//   } else {
-//     console.error('Apply filters button not found')
-//   }
-//   setSelectedAnimalType()
-// })
+document.addEventListener('DOMContentLoaded', function () {
+  // Event listeners for animal filter buttons
+  document.querySelectorAll('#singleAnimalFilter div[data-type]').forEach(selectingDivFilter => {
+    selectingDivFilter.addEventListener('click', function () {
```
Comment (vladiiancu) [inline public/JavaScript/gallery.js]: userSelectedFilters, activeFilters, galleryItems are great. 

Names such as selectingDivFilter not so much. What is a div? If I change the html from a div to a span will this name still make sense?

**Review by vladiiancu — COMMENTED** [review]
Now, please make the necessary changes for the tests to pass. It's more around organizing the code into files that anything.

File: public/gallery.html [hunk 4]
```diff
@@ -262,5 +262,6 @@
 
 <script src="JavaScript/common.js"></script>
 <script src="JavaScript/gallery.js"></script>
```
Comment (vladiiancu) [inline public/gallery.html:264]: Check out https://www.sitepoint.com/using-es-modules/. Using ES modules inside the browser means that you could add only the relevant script here and then import anything else needed in the file. 
What do you think? Would you like to make this change in a separate PR? I will leave it up to you

Comment (Alexandru-Bulai) [reply public/gallery.html:264]: i see.... i will have a look now

File: public/gallery.html [hunk 5]
```diff
@@ -261,6 +261,7 @@
 <!-- Images gallery container end -->
 
 <script src="JavaScript/common.js"></script>
-<script src="JavaScript/gallery.js"></script>
+<script type="module" src="JavaScript/gallery.js"></script>
+<script type="module" src="JavaScript/gallery-dom.js"></script>
```
Comment (vladiiancu) [inline public/gallery.html]: If we are using modules, then you don't need both of them here. You only need the initial one. And then the initial one does the import of the other file. 

HOWEVER, I forgot that jest doesn't play nice with modules for now. So the unit tests fails.
Revert to non modules please.

**Review by vladiiancu — CHANGES_REQUESTED** [review]

**Review by vladiiancu — APPROVED** [review]

**Pull request merged**